src/SelfService/Application/KafkaTopicApplicationService.cs

```diff
@@ -24,6 +26,205 @@ IKafkaTopicRepository kafkaTopicRepository
         _kafkaTopicRepository = kafkaTopicRepository;
     }
 
+    public async Task CheckIfCanRequestContract(
+        KafkaTopicId kafkaTopicId,
+        MessageType messageType,
+        MessageContractSchema newSchema
+    )
+    {
+        var schemaVersion = newSchema.GetSchemaVersion();
+        if (schemaVersion == null)
+            throw new ArgumentException("Cannot request new message contract without schema version");
+
+        var contracts = (await _messageContractRepository.FindBy(kafkaTopicId, messageType)).ToArray();
+        if (contracts.Length == 0)
+        {
+            if (schemaVersion != 1)
+                throw new ArgumentException("Cannot request new message contract with schema version other than 1");
+            return;
+        }
+
+        var latestSchemaVersion = 0;
+        MessageContract? latestContract = null;
+        foreach (MessageContract contract in contracts)
+        {
+            if (contract.SchemaVersion > latestSchemaVersion)
+            {
+                latestSchemaVersion = contract.SchemaVersion;
+                latestContract = contract;
+            }
+
+            if (contract.SchemaVersion != schemaVersion)
+                continue;
```

Comment:
This continue seems like it should do something else than just continue.
This is an unwanted situation, is it not?

Reply:
rewrote to get latest schema